src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -578,117 +669,137 @@ class HookUtils:
-        pbs.logmsg(pbs.EVENT_DEBUG2, "%s: Host assigned job resources: %s" %
+        # Avoid printing repr by default due to its size
+        # pbs.logmsg(pbs.EVENT_DEBUG4, "%s: %s" % (caller_name(), repr(node)))
+        pbs.logmsg(pbs.EVENT_DEBUG4, "%s: Host assigned job resources: %s" %
                    (caller_name(), jobutil.assigned_resources))
-        with lock(cgroup.cfg['cgroup_lock_file']):
+        with Lock(cgroup.cfg['cgroup_lock_file']):
             # Make sure the parent cgroup directories exist
             cgroup.create_paths()
             # Make sure the cgroup does not already exist
             # from a failed run
-            cgroup.delete(e.job.id, False)
+            cgroup.delete(event.job.id, False)
             # Create the cgroup(s) for the job
-            cgroup.create_job(e.job.id, node)
+            cgroup.create_job(event.job.id, node)
             # Configure the new cgroup
-            cgroup.configure_job(e.job.id, jobutil.assigned_resources, node)
+            cgroup.configure_job(event.job.id,
+                                 jobutil.assigned_resources, node)
             # Initialize resource usage for the job
-            cgroup.update_job_usage(e.job.id, e.job.resources_used)
+            cgroup.update_job_usage(event.job.id, event.job.resources_used)
             # Write out the assigned resources
-            cgroup.write_out_cgroup_assigned_resources(e.job.id)
+            cgroup.write_cgroup_assigned_resources(event.job.id)
             # Write out the environment variable for the host (pbs_attach)
             if 'device_names' in cgroup.assigned_resources:
-                pbs.logmsg(pbs.EVENT_DEBUG2, "%s: Devices: %s" %
+                pbs.logmsg(pbs.EVENT_DEBUG4, "%s: Devices: %s" %
                            (caller_name(),
                             cgroup.assigned_resources['device_names']))
                 env_list = []
-                if len(cgroup.assigned_resources['device_names']) > 0:
+                if cgroup.assigned_resources['device_names']:
                     mics = []
                     gpus = []
                     for key in cgroup.assigned_resources['device_names']:
                         if key.startswith('mic'):
                             mics.append(key[3:])
                         elif key.startswith('nvidia'):
                             gpus.append(key[6:])
-                    if len(mics) > 0:
+                    if mics:
                         env_list.append('OFFLOAD_DEVICES="%s"' %
                                         string.join(mics, ','))
-                    if len(gpus) > 0:
+                    if gpus:
                         # Don't put quotes around the values. ex "0" or "0,1".
                         # This will cause it to fail.
                         env_list.append('CUDA_VISIBLE_DEVICES=%s' %
                                         string.join(gpus, ','))
```

Comment:
Please, test the settings of CUDA_VISIBLE_DEVICES with latest master branch. There are some (recent) incompatible changes. I think It is necessary to escape the comma with \\ due to recent changes in #393. Similar, you need to escape comma and remove quotation marks on [this line](https://github.com/mike0042/pbspro/blob/0c536e3fda0cce69dde25e53ebe002e3a209f1e2/src/hooks/cgroups/pbs_cgroups.PY#L2314)

Vasek

Reply:
Hello @vchlum, I don't currently have any CUDA devices to test with so I'll need your assistance if you don't mind. I'll add your suggested changes to the next commit and ask that you test them. Do you think we need the same changes for "offload_devices" as well?

Reply:
@mike0042 Sure, I will test it on monday because I need to wait for one of the node with CUDA to be fully drained. Concerning the offload_devices, I think you need to escape the comma too, but I don't know whether the variable OFFLOAD_DEVICES is allowed to contain quotation marks or not.

Reply:
Wait... Maybe this is a hack - to escape the comma. The correct way is probably to fix it somewhere else... What do you think? The hook should not be aware of the necessity to escape the value of the variable.

Reply:
I know there were some recent changes related to environment variables containing "special characters" (e.g. commas, spaces, carriage returns, etc.). It's conceivable that this has already been fixed. I'm going to leave the code as-is for now. We can always file another ticket if need be. Thanks for your input.